Ignore subscriptshift/superscriptshift when MathML Core is enabled

eerii opened the pull request:
#### 4077efe2bdb9af282c1638b6294af56d0ca964f8
<pre>
Ignore subscriptshift/superscriptshift when MathML Core is enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=281208">https://bugs.webkit.org/show_bug.cgi?id=281208</a>

Reviewed by Frédéric Wang.

Return a default Length for subscriptShift and superscriptShift when the
CoreMathML preference is on.

Tests: mathml/non-core/msub-subscriptshift.html
       mathml/non-core/msubsup-subscriptshift.html
       mathml/non-core/msubsup-superscriptshift.html
       mathml/non-core/msup-superscriptshift.html
       mathml/non-core/multiscript-subscriptshift.html
       mathml/non-core/multiscript-superscriptshift.html
* LayoutTests/TestExpectations:
* LayoutTests/mathml/non-core/msub-subscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msub-subscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msub-subscriptshift.html: Renamed from LayoutTests/mathml/presentation/msub-subscriptshift.html.
* LayoutTests/mathml/non-core/msubsup-subscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msubsup-subscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msubsup-subscriptshift.html: Renamed from LayoutTests/mathml/presentation/msubsup-subscriptshift.html.
* LayoutTests/mathml/non-core/msubsup-superscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msubsup-superscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msubsup-superscriptshift.html: Renamed from LayoutTests/mathml/presentation/msubsup-superscriptshift.html.
* LayoutTests/mathml/non-core/msup-superscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msup-superscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msup-superscriptshift.html: Renamed from LayoutTests/mathml/presentation/msup-superscriptshift.html.
* LayoutTests/mathml/non-core/multiscript-subscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/multiscript-subscriptshift-expected.txt.
* LayoutTests/mathml/non-core/multiscript-subscriptshift.html: Renamed from LayoutTests/mathml/presentation/multiscript-subscriptshift.html.
* LayoutTests/mathml/non-core/multiscript-superscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/multiscript-superscriptshift-expected.txt.
* LayoutTests/mathml/non-core/multiscript-superscriptshift.html: Renamed from LayoutTests/mathml/presentation/multiscript-superscriptshift.html.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/mathml/MathMLScriptsElement.cpp:
(WebCore::MathMLScriptsElement::subscriptShift):
(WebCore::MathMLScriptsElement::superscriptShift):
* Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp:
(WebCore::RenderMathMLScripts::verticalMetrics):

Canonical link: <a href="https://commits.webkit.org/300817@main">https://commits.webkit.org/300817@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39fba25b26e1e9732124c2e3cad8f1fc220204cf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123940 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43655 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34352 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130731 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125817 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44379 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52249 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94272 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126894 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35359 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110869 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74873 "Found 1 new API test failure: TestWTF:WTF_Condition.OneProducerOneConsumerOneSlotTimeout (failure)") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/34322 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29031 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74203 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105089 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29253 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133423 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50893 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38764 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102742 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51267 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107089 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102566 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26093 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47911 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26161 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47746 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50746 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56510 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50220 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53566 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51894 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fred-wang commented on Source/WebCore/mathml/MathMLScriptsElement.cpp:
```diff
@@ -67,13 +68,19 @@ Ref<MathMLScriptsElement> MathMLScriptsElement::create(const QualifiedName& tagN
+
 const MathMLElement::Length& MathMLScriptsElement::subscriptShift()
 {
+    if (document().settings().coreMathMLEnabled())
```
It seems these are only used here:
https://searchfox.org/wubkat/rev/ae89fb172f2f6dfa362aff8493e0e2978a9800d4/Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp#281
https://searchfox.org/wubkat/rev/ae89fb172f2f6dfa362aff8493e0e2978a9800d4/Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp#290

How about we skip these branches when MathML Core is enabled, and add debug assertions in MathMLScriptsElement::subscriptShift()/MathMLScriptsElement::superscriptShift() to make sure they are never called when the pref is enabled?

(this would avoid allocating this defaultLength constant)

eerii replied:
That sounds better, thank you

fred-wang commented on LayoutTests/mathml/non-core/msub-subscriptshift.html:
Can you please move this and similar tests to LayoutTests/mathml/non-core ?

eerii replied:
Ah yes, sorry for not realizing before

eerii committed:
Ignore subscriptshift/superscriptshift when MathML Core is enabled
https://bugs.webkit.org/show_bug.cgi?id=281208

Reviewed by Frédéric Wang.

Return a default Length for subscriptShift and superscriptShift when the
CoreMathML preference is on.

Tests: mathml/non-core/msub-subscriptshift.html
       mathml/non-core/msubsup-subscriptshift.html
       mathml/non-core/msubsup-superscriptshift.html
       mathml/non-core/msup-superscriptshift.html
       mathml/non-core/multiscript-subscriptshift.html
       mathml/non-core/multiscript-superscriptshift.html
* LayoutTests/TestExpectations:
* LayoutTests/mathml/non-core/msub-subscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msub-subscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msub-subscriptshift.html: Renamed from LayoutTests/mathml/presentation/msub-subscriptshift.html.
* LayoutTests/mathml/non-core/msubsup-subscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msubsup-subscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msubsup-subscriptshift.html: Renamed from LayoutTests/mathml/presentation/msubsup-subscriptshift.html.
* LayoutTests/mathml/non-core/msubsup-superscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msubsup-superscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msubsup-superscriptshift.html: Renamed from LayoutTests/mathml/presentation/msubsup-superscriptshift.html.
* LayoutTests/mathml/non-core/msup-superscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/msup-superscriptshift-expected.txt.
* LayoutTests/mathml/non-core/msup-superscriptshift.html: Renamed from LayoutTests/mathml/presentation/msup-superscriptshift.html.
* LayoutTests/mathml/non-core/multiscript-subscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/multiscript-subscriptshift-expected.txt.
* LayoutTests/mathml/non-core/multiscript-subscriptshift.html: Renamed from LayoutTests/mathml/presentation/multiscript-subscriptshift.html.
* LayoutTests/mathml/non-core/multiscript-superscriptshift-expected.txt: Renamed from LayoutTests/mathml/presentation/multiscript-superscriptshift-expected.txt.
* LayoutTests/mathml/non-core/multiscript-superscriptshift.html: Renamed from LayoutTests/mathml/presentation/multiscript-superscriptshift.html.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/mathml/MathMLScriptsElement.cpp:
(WebCore::MathMLScriptsElement::subscriptShift):
(WebCore::MathMLScriptsElement::superscriptShift):
* Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp:
(WebCore::RenderMathMLScripts::verticalMetrics):

Canonical link: https://commits.webkit.org/300817@main
